src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/CredentialStore.php

```diff
@@ -58,120 +52,96 @@ public static function getInstance()
     }
 
     /**
-     * CredentialStore constructor.
-     */
-    private function __construct()
-    {
-        $this->encodedKey = base64_encode(openssl_random_pseudo_bytes(16));
-        $this->iv = substr(hash('sha256', $this->encodedKey), 0, 16);
-        $creds = $this->readInCredentialsFile();
-        $this->credentials = $this->encryptCredFileContents($creds);
-    }
-
-    /**
-     * Returns the value of a secret based on corresponding key
+     * CredentialStore constructor
      *
-     * @param string $key
-     * @return string|null
      * @throws TestFrameworkException
      */
-    public function getSecret($key)
+    private function __construct()
     {
-        if (!array_key_exists($key, $this->credentials)) {
-            throw new TestFrameworkException(
-                "{$key} not defined in .credentials, please provide a value in order to use this secret in a test."
-            );
+        // Initialize vault storage
```

Comment:
Make sure to initialize file storage first, since it's the prefered method.

Reply:
Changed.